### PR TITLE
Make `type_erased_stream` `Scheduler` aware

### PR DESCRIPTION
**Pull request by janondrusek**
* expose `unifex::type_erased_stream` (result of `unifex::type_erase()`)
* add `get_scheduler` CPO into `type_erased_stream`

### CODE REVIEW

File: include/unifex/type_erased_stream.hpp [hunk 1]
```diff
@@ -107,18 +126,23 @@ struct _stream<Values...>::type {
       continuation_info get_continuation_info() const noexcept override {
         return continuation_info::from_continuation(receiver_);
       }
+
+      any_scheduler_ref get_scheduler() const noexcept override {
+        return scheduler_;
```
Comment (MarkCMann) [inline include/unifex/type_erased_stream.hpp:131]: Do you need the scheduler to be a member variable? Seems that you could delay evaluation of `get_scheduler` until this point and save on some byes.

File: include/unifex/type_erased_stream.hpp [hunk 2]
```diff
@@ -132,22 +156,26 @@ struct _stream<Values...>::type {
...
```
Comment (MarkCMann) [inline include/unifex/type_erased_stream.hpp:161]: Same question as above

File: include/unifex/type_erased_stream.hpp (hunk 1 above)
Comment (MarkCMann) [reply include/unifex/type_erased_stream.hpp:131]: get_scheduler returns schedulers by value in all of the other cases, so this class needs to store the downstream receivers scheduler by value as a member to ensure that the scheduler reference passed to any_scheduler_ref lives beyond the scope of this function call.

Comment (janondrusek) [reply include/unifex/type_erased_stream.hpp:131]: Although not documented as a requirement: all existing `get_scheduler()` implementations return by value. If we returned inline, `any_scheduler_ref` would take a temporary.

**Review by ispeters — APPROVED** [review]
I'm assuming you've rebased onto the recent `task<>` fix so lgtm.

**Pull request merged**